### PR TITLE
#18 removed allocation of Stopwatch and replaced with two longs

### PR DESCRIPTION
**Pull request by EamonHetherton**

### CODE REVIEW

**Comment by EamonHetherton** [comment]
There was only one simple open issue; I couldn't resist :)

File: src/SerilogTimings/Operation.cs [hunk 1]
```diff
@@ -81,14 +82,14 @@ internal Operation(ILogger target, string messageTemplate, object[] args, Comple
...
-            _stopwatch = Stopwatch.StartNew();
+            _startTicks = Stopwatch.GetTimestamp();
         }
 
         /// <summary>
         /// Returns the elapsed time of the operation. This will update during the operation, and be frozen once the
         /// operation is completed or canceled.
         /// </summary>
-        public TimeSpan Elapsed => _stopwatch.Elapsed;
+        public TimeSpan Elapsed => _stopTicks == 0 ? TimeSpan.FromTicks(Stopwatch.GetTimestamp() - _startTicks) : TimeSpan.FromTicks(_stopTicks - _startTicks);
```
Comment (nblumhardt) [inline src/SerilogTimings/Operation.cs]: Stopwatch ticks and timespan ticks are not at the same resolution, sadly.

Comment (EamonHetherton) [reply src/SerilogTimings/Operation.cs]: Didn't know abouth the ticks thing and reading through #20 I see you've been though this before :)  I'll dig into Stopwatch and see if I can get this resolved, but wont have time for another week, but given its history, who's counting :)

**Comment by nblumhardt** [comment]
Picking this up now 😅  -- hope 2022 is off to a good start, @EamonHetherton !

**Comment by nblumhardt** [comment]
On NuGet.org now, though with a few TFM compatibility changes; let me know if it looks like I broke too much in my last-minute hacking 😅

**Comment by nblumhardt** [comment]
Sorry, I spoke too soon, build issue yet to fix..

**Comment by davidpeden3** [comment]
@nblumhardt i literally just needed this again and walked through all of our chat history in the other two PRs. what's the status on this? how do i get my beloved `.Elapsed`?  🥺

**Comment by nblumhardt** [comment]
Hi David! How does version `3.0.0-dev-*` from NuGet work for you? It might be time to mark 3.0.0 as done 🤔

**Comment by nblumhardt** [comment]
Looks like a nice improvement @davidpeden3, if you're happy to lose the "floor" functionality 👍 

Might be a bit of a feature gap at this stage - beyond a certain point it'll be better to just write the extra code outside this library than to try to incorporate more into the API.

**Comment by davidpeden3** [comment]
reading into your comment, i will not submit a pr to add the capability. thanks for the feedback. when will 3.0 move out of pre-release?

image

**Comment by nblumhardt** [comment]
Queued up https://github.com/nblumhardt/serilog-timings/pull/46 - thanks @davidpeden3

**Pull request merged**